src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/SyncPeerProtocolHandlerBase.cs

```diff
@@ -51,6 +53,7 @@ public abstract class SyncPeerProtocolHandlerBase : ZeroProtocolHandlerBase, ISy
 
         protected readonly MessageQueue<GetBlockHeadersMessage, BlockHeader[]> _headersRequests;
         protected readonly MessageQueue<GetBlockBodiesMessage, BlockBody[]> _bodiesRequests;
+        protected LruKeyCache<Keccak> NotifiedTransactions { get; } = new(MemoryAllowance.MemPoolSize, "notifiedTransactions");
```

Comment:
This is 2048 total; and any freshly connected peers get sent all hashes, so this is just reducing the delta?

Reply:
i.e. does it only need one LruCache for the protocol rather than per peer?

Reply:
Not sure what is the question, but we need to track to whom we send what hashes.

Reply:
> i.e. does it only need one LruCache for the protocol rather than per peer for a shared delta?

AH got it, generally depends, but we only gossip transaction in eth protocol, so only one is needed for now.